libs/s25main/ingameWindows/IngameWindow.cpp

```diff
@@ -45,8 +47,9 @@ IngameWindow::IngameWindow(unsigned id, const DrawPoint& pos, const Extent& size
     // Load last position or center the window
     if(pos == posLastOrCenter)
     {
-        if(id < last_pos.size() && last_pos[id].isValid())
-            SetPos(last_pos[id]);
+        auto& settings = SETTINGS.windows.persistentSettings;
```

Comment:
It doesn't work exactly the way you suggested, because the []-operator for maps is not const. But I now changed it to use the method std::map::find() and then using iterators, which allows this to be const.